packages/template-retail-react-app/app/utils/jwt-utils.js

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {createRemoteJWKSet as joseCreateRemoteJWKSet, jwtVerify} from 'jose'
+import {getAppOrigin} from '@salesforce/pwa-kit-react-sdk/utils/url'
+import {getConfig} from '@salesforce/pwa-kit-runtime/utils/ssr-config'
+
+const throwSlasTokenValidationError = (message, code) => {
+    throw new Error(`SLAS Token Validation Error: ${message}`, code)
+}
+
+export const createRemoteJWKSet = () => {
+    const appOrigin = getAppOrigin()
+    const {app: appConfig} = getConfig()
+    const shortCode = appConfig.commerceAPI.parameters.shortCode
+    const tenantId = appConfig.commerceAPI.parameters.organizationId.replace(/^f_ecom_/, '')
+    const JWKS_URI = `${appOrigin}/${shortCode}/${tenantId}/oauth2/jwks`
+    return joseCreateRemoteJWKSet(new URL(JWKS_URI))
+}
+
+export const validateSlasCallbackToken = async (token) => {
+    try {
+        const jwks = createRemoteJWKSet()
+        const {payload} = await jwtVerify(token, jwks, {})
```

Comment:
Are we only relying on `jwtVerify()` to validate the token? Should we also check headers, such as `kid`, `alg`, or `jku`?

I'm not an expert in this area, but I recommend reaching out to a SLAS expert who can guide you on the what is the CC standard to consider a token valid. 

Another question for the SLAS expert is whether it is safe to share these validation logic in a public repo.

Reply:
@adamraya I believe it is sufficient to call `jwtVerify` and have the validation logic in this repo. Unless the PWA team has preferences on where this logic is placed, security wise I believe it is not a risk.

From Blair: "There is no need to validate the header portion of the JWT. The kid is the most important header claim as that is the KeyID of the public key to validate the body signature. Calling jwtVerify should be fine.
I think it would be okay to add the validation of the JWT signature to the repo. After all it is using the Public keys and not exposing the private key or how the JWT was signed."

Reply:
I see. I mention this because we usually have an additional layer of validation that checks those claims to ensure the token context is valid, which `jwtVerify` does not validate. But let's continue with what the experts say.